pixano/data/settings.py

```diff
@@ -11,17 +11,64 @@
 #
 # http://www.cecill.info
 
-
 from pathlib import Path
+from typing import Optional
+from urllib.parse import urlparse
 
+import boto3
 from pydantic_settings import BaseSettings
+from s3path import S3Path, register_configuration_parameter
 
 
 class Settings(BaseSettings):
     """Dataset library settings
 
     Attributes:
-        data_dir (Path): Dataset library directory
+        data_dir (Path | S3Path): Dataset library directory, or url to S3 path (eg. "s3://<bucket>[/<key>]")
+
+        If data_dir is a S3 url, the following environment variables can be provided
```

Comment:
This line still needs to be removed for the documentation website to compile the docstring correctly. Otherwise the attributes below it are not registered.

Other than that, all good!